react/__tests__/LocationSearch.test.tsx

```diff
@@ -1,18 +1,53 @@
 import React from 'react'
 import { render, screen, fireEvent } from '@vtex/test-tools/react'
+import { MockedProvider, MockedResponse } from '@apollo/react-testing'
+import { Address } from 'vtex.checkout-graphql'
+import { AddressRules } from 'vtex.address-context/types'
 
 import LocationSearch from '../LocationSearch'
-import { getFewAddresses } from '../__fixtures__/fixtures'
+import { AddressContextProvider } from '../__mocks__/vtex.address-context/AddressContext'
+import { googleLogo } from '../__fixtures__/graphql/providerLogo.fixture'
+import {
+  noSuggestions,
+  simpleSuggestions,
+} from '../__fixtures__/graphql/suggestAddresses.fixture'
+import { simpleSuggestedAddress } from '../__fixtures__/graphql/getAddressByExternalId.fixture'
 
 const UNTABABBLE = -1
+const LABEL_MATCHER = /Address \(street, number, square\.\.\.\)/i
+
+interface RenderComponentProps {
+  address?: Address | null
+  countries?: string[]
+  rules?: AddressRules
+  graphqlMocks?: MockedResponse[]
+}
 
 describe('Location Search', () => {
+  const renderComponent = ({
+    address = null,
+    countries = [],
+    rules = {},
+    graphqlMocks = [],
+  }: RenderComponentProps) => {
+    return render(
+      <MockedProvider addTypename={false} mocks={graphqlMocks}>
+        <AddressContextProvider
+          address={address}
+          countries={countries}
+          rules={rules}
+        >
+          <LocationSearch />
+        </AddressContextProvider>
+      </MockedProvider>
+    )
+  }
   it('should be able to select a suggested address when a search term is found', async () => {
-    render(<LocationSearch getAddresses={getFewAddresses} />)
+    renderComponent({
+      graphqlMocks: [googleLogo, simpleSuggestions, simpleSuggestedAddress],
+    })
 
-    const input = screen.getByLabelText(
-      /Address and house number/i
-    ) as HTMLInputElement
+    const input = screen.getByLabelText(LABEL_MATCHER) as HTMLInputElement
```

Comment:
why do you need this cast?

Reply:
Otherwise it returns a HTMLElement. But tbh, I don't remember why that was relevant

Reply:
Just removed all the occurrences and everything works fine. Do you think I should remove it?

Reply:
yeah, looks better to me